### PR TITLE
Catwm models

### PR DESCRIPTION
**Pull request by CatW-M**

### CODE REVIEW

**Review by cormacpujals — APPROVED** [review]
Wow, optional chaining and .lean().. super cool. I look forward to learning from your backend skills! 

I'm going to approve on my end, don't know if that will allow Nathan to review it before final merge.. we'll see.

**Review by nghiavo24 — APPROVED** [review]
Thank you for very descriptive codes. I learned a lot!!!

**Pull request merged**
